### PR TITLE
refactor: minor perf improvements and typing changes

### PR DESCRIPTION
**Pull request by micalevisk**
see the commit messages

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 lockfileVersion: '6.0'
 
+settings:
+  autoInstallPeers: true
+  excludeLinksFromLockfile: false
+
```
Comment (micalevisk) [inline pnpm-lock.yaml:6]: I guess I need to revert that, right?

this was indroduced after running `pnpm remove rimraf`
```suggestion
```

File: src/exploration.module.ts [hunk 2]
```diff
@@ -55,31 +55,36 @@ export class ExplorationModule {
...
-      .filter((module) => module.metatype.name !== InternalCoreModule.name)
-      .map((module) => module.metatype.name);
+    // NOTE: Using 'forEach' here instead of filter+map for performance reasons.
+    const importsNames: string[] = [];
+    module.imports.forEach((importedModule) => {
+      if (importedModule.metatype.name !== InternalCoreModule.name) {
+        importsNames.push(importedModule.metatype.name);
+      }
+    });
+    return importsNames;
```
Comment (jmcdo29) [inline src/exploration.module.ts:65]: If we're looking at true performance reasons, shuoldn't we be using `for of`?

File: src/exploration.module.ts [hunk 3]
```diff
@@ -105,20 +110,24 @@ export class ExplorationModule {
...
 
   private static getExports(module: NestModule): string[] {
-    return Array.from(module.exports).map((exportValue) =>
-      this.getInjectionToken(exportValue),
-    );
+    const exportsNames: string[] = [];
+    module.exports.forEach((exportValue) => {
+      exportsNames.push(this.getInjectionToken(exportValue));
+    });
+    return exportsNames;
```
Comment (jmcdo29) [inline src/exploration.module.ts:130]: `for of`

File: pnpm-lock.yaml (hunk 1 above)
Comment (jmcdo29) [reply pnpm-lock.yaml:6]: Eh, this is probably just pnpm doing what pnpm needs to do. I think this was s difference back on v6 vs current v8?

**Review by jmcdo29 — COMMENTED** [review]
A few other spots where we can use `for or`

File: pnpm-lock.yaml (hunk 1 above)
Comment (micalevisk) [reply pnpm-lock.yaml:6]: probably. I'm using v8

**Review by jmcdo29 — APPROVED** [review]

**Comment by jmcdo29** [comment]
@micalevisk all looks good, ready for me to merge?

**Comment by micalevisk** [comment]
@jmcdo29 yep

**Pull request merged**